src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -756,9 +756,14 @@ private static IntermediateNode GetAddBinaryOp(IRTranslatorContext context, Texl
                             // Date + '-DateTime' => in days
                             // Date + '-Date' => in days
 
-                            // Ensure that this is really '-Date' - Binding should always catch this, but let's make sure...
-                            Contracts.Assert(node.Right.AsUnaryOpLit().VerifyValue().Op == UnaryOp.Minus);
-                            return new BinaryOpNode(context.GetIRContext(node), BinaryOpKind.DateDifference, left, right);
+                            // Binding produces this as '-Date'. This should be cleaned up when we switch to a proper sub op. 
+                            if (right is not UnaryOpNode unaryNode || unaryNode.Op != UnaryOpKind.Negate)
+                            {
+                                throw new NotSupportedException();
```

Comment:
>throw new NotSupportedException(); [](http://example.com/codeflow?start=32&length=34)

Is there a legal expression that can land here?

Or will binding have prevented it and landing here is a compiler bug.

Reply:
Binding will have prevented this. It's always a bug if we land here.

Reply:
see 
![image](https://user-images.githubusercontent.com/69215460/165846075-b7dee30e-31ee-47d2-a2ca-cc87ffe5714a.png)